### PR TITLE
Publicize: Show "Share this post" panel open if there are publicize connections enabled

### PR DESCRIPTION
**Pull request by oskosk**
#### Changes proposed in this Pull Request:

* Sets `initialOpen={true}` if there are publicize connections enabled

#### After

Screen Shot 2022-10-06 at 08 31 29

#### Before

Screen Shot 2022-10-06 at 08 28 48

#### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?

#### Does this pull request change what data or activity we track or use?

No

#### Testing instructions:

* Without a social connection: Attempt to publish a post and confirm the "Share this post" panel is closed
* Enable one connection: Attempt to publish a post and confirm the "Share this post" panel is open

### CODE REVIEW

File: projects/plugins/jetpack/extensions/plugins/publicize/index.js [hunk 1]
```diff
@@ -9,19 +9,22 @@
  */
 
 import { JetpackLogo } from '@automattic/jetpack-components';
-import { TwitterThreadListener } from '@automattic/jetpack-publicize-components';
+import {
+	TwitterThreadListener,
+	useSocialMediaConnections as useSelectSocialMediaConnections,
```
Comment (danielpost) [inline projects/plugins/jetpack/extensions/plugins/publicize/index.js]: Just out of curiosity, why did you rename this hook?

**Review by danielpost — DISMISSED** [review]
Works as advertised, thanks for picking this up!

File: projects/plugins/jetpack/extensions/plugins/publicize/index.js (hunk 1 above)
Comment (oskosk) [reply projects/plugins/jetpack/extensions/plugins/publicize/index.js]: copypasta

Comment (oskosk) [reply projects/plugins/jetpack/extensions/plugins/publicize/index.js]: I'll remove

Comment (oskosk) [reply projects/plugins/jetpack/extensions/plugins/publicize/index.js]: Addressed in 19baa835576d2c002ec9d97ce0909517c6ba082b

**Review by danielpost — APPROVED** [review]

**Review by nunyvega — APPROVED** [review]
